pages/en/contrib/style-guide.md

```diff
@@ -24,6 +25,152 @@ Style conventions for this document:
 
 ## General guidelines
 
+{% include important.html content="
+
+The rules described in this document are evolving, therefore not all of our
+code-base is following them yet. This is expected and OK. Our approach is to
+incrementally improve the parts that are changing most often, i.e. fix the
+coding style only as part of changes needed for a bug fix/new feature.
+
+" %}
+
+### Variable declarations
+
+Prefer to use immutable variables declared with `const` keyword:
+
+### Good
+
+```js
+const user = app.models.User;
+const saltWorkFactor = user.settings.saltWorkFactor;
+```
+
+#### Bad
+
+```js
+var user = app.models.User;
+var saltWorkFactor = user.settings.saltWorkFactor;
+```
+
+In the rare cases where you need to split variable declaration and
+initialization, prefer to use `let` over `var`:
+
+```js
+let discount = 0;
+
+if (customer.isPreferred) {
+  discount = 0.1;
+} else if (customer.address.state = 'CA') {
+  discount = 0.05;
+}
+```
+
+In most cases, it's possible to rewrite such code by extracting the
+conditional logic into a standalone function, which is preferred:
+
+```js
+const discount = customer.getDiscount();
+
+// Inside Customer model:
+customer.prototype.getDiscount = function() {
+  if (customer.isPreferred) {
+    return 0.1;
+  }
+
+  if (customer.address.state = 'CA') {
+    return 0.05;
+  }
+
+  return 0;
+}
+```
+
+### Arrow functions
+
+The are two considerations to keep in mind when deciding whether to use an
+arrow function or a regular function:
+
+ - Arrow function preserves `this` from the outer scope. This is much more
+   perfomant than the usual workaround of storing `this` in a `self` variable.
+
+ - If you are accessing `this` provided by Mocha to your test cases and
+   before/after hooks, then you cannot use an arrow function.
+
+ - Arrow functions are anonymous, there is no easy way for assigning them a
+   descriptive name. This makes it difficult to understand stack traces.
```

Comment:
Half-true.  V8 does a fairly good job of inferring a name when you assign them to variables.